pubsub/requests.go

```diff
@@ -35,3 +43,26 @@ type NewMessage struct {
 	Metadata    map[string]string `json:"metadata"`
 	ContentType *string           `json:"contentType,omitempty"`
 }
+
+// BulkMessage represents bulk message arriving from a message bus instance
+type BulkMessage struct {
+	Entries  []BulkMessageEntry `json:"entries"`
+	Topic    string             `json:"topic"`
+	Metadata map[string]string  `json:"metadata"`
+}
+
+// BulkMessageEntry represents a single message inside a bulk request
+type BulkMessageEntry struct {
+	EntryID     string            `json:entryID`
+	Event       []byte            `json:"event"`
+	ContentType string            `json:"contentType,omitempty"`
+	Metadata    map[string]string `json:"metadata"`
+}
+
+// BulkSubscribeConfig defines the Configurations that can be applied to control bulk subscribe
+// behavior - beahvior may depend per building block
+type BulkSubscribeConfig struct {
+	MaxBulkCount               int `json:"maxBulkCount"`
+	MaxBulkLatencyMilliSeconds int `json:"maxBulkLatencyMilliSeconds"`
```

Comment:
```suggestion
	MaxBulkAwaitDurationMilliSeconds int `json:"maxBulkAwaitDurationMilliSeconds"`
```
This is similar to [RocketMQ](https://www.alibabacloud.com/help/en/message-queue-for-apache-rocketmq/latest/batch-consumption), and it sounds more intuitive than latency.